lua/luasnip/init.lua

```diff
@@ -61,19 +61,22 @@ local function get_context(snip)
 	}
 end
 
-local function available()
+local function available(opts)
+	opts = opts or {}
+	opts.get_context = opts.get_context or get_context
```

Comment:
`get_context` could be immediately stored in a local variable here, no need for the access through `opts`

Reply:
Ohhh I see xD
`effective_get_context` would be nice (or change the current `get_context` to `default_get_context`, or something more descriptive (`docstring_context`?? since that's the difference to the original `get_context`))

Reply:
So as we discussed below, if you're ok with it, I can rename the original get_context to ```default_snip_info``` and the available function would look like

```lua
local function available(snip_info)
snip_info = default_snip_info or snip_info
end
```

Reply:
Yeah that sounds good :+1:

Reply:
Solved in https://github.com/L3MON4D3/LuaSnip/pull/652/commits/868416192026216a556b62950c0463734b07f388